File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    i = 0
    while True:
        cnt = 0
        
        for idx, v in enumerate(arr):
            if v >= 50 and v%2 == 0:
                arr[idx] //= 2
            elif v < 50 and v%2 == 1:
                arr[idx] = arr[idx]*2 + 1
            else:
                cnt += 1
        
        if cnt == len(arr):
            return i
        
        i += 1
        
                
        
                

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.*;

class Solution {
    public int solution(int[] arr) {
        int cnt = 0;
        while (true) {
            int[] newArr = Arrays.copyOf(arr, arr.length);
            boolean flag = false;
            for (int i = 0; i < arr.length; i++) {
                if (arr[i] >= 50 && arr[i] % 2 == 0) {
                    newArr[i] = newArr[i] / 2;
                    flag = true;
                } else if (arr[i] < 50 && arr[i] % 2 != 0) {
                    newArr[i] = arr[i] * 2 + 1;
                    flag = true;
                }
            }
            if (!flag) {
                return cnt;
            }

            arr = newArr;
            cnt += 1;
        }
    }
}